.github/workflows/integration-testing.yml

```diff
@@ -0,0 +1,194 @@
+        with:
+          java-version: 14
+
+      - uses: gradle/wrapper-validation-action@v1
+
+      - name: Build with Gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          arguments: build --stacktrace -PenableCoverage=true
+
+      - name: Upload agent artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: aws-opentelemetry-agent.jar
+          path: otelagent/build/libs/aws-opentelemetry-agent-*.jar
+
+  build_Spring_App:
+    name: Build Spring App
+    needs: build_Java_Agent
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 14
+
+      - uses: gradle/wrapper-validation-action@v1
+
+      - name: Build Spring boot web app with Gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          arguments: build --stacktrace -PenableCoverage=true
+          build-root-directory: sample-apps/springboot
+
+      - name: Upload web app build artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: springapp.jar
+          path: sample-apps/springboot/build/libs/springboot-1.0.jar
+
+  build_Spark_App:
+    name: Build Spark App
+    needs: build_Java_Agent
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 14
+
+      - uses: gradle/wrapper-validation-action@v1
+
+      - name: Build Spark web app with Gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          arguments: build --stacktrace -PenableCoverage=true
+          build-root-directory: sample-apps/spark
+          gradle-version: 6.5.1
+
+      - name: Upload web app build artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: sparkapp.jar
+          path: sample-apps/spark/build/libs/sparkapp.jar
+
+  test_Spring_With_Java_Agent:
+    name: Test Spring App with AWS OTel Java agent
+    needs: build_Spring_App
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download AWS OTel Jave agent jar
+        uses: actions/download-artifact@v2
+        with:
+          name: aws-opentelemetry-agent.jar
+
+      - name: Download Spring Boot App jar
+        uses: actions/download-artifact@v2
+        with:
+          name: springapp.jar
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 14
+
+      - name: Setup and Run OTel Collector
+        run: |
+          docker-compose -f .github/collector/docker-compose.yml up &
+        env:
+          ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      - name: Run Spring Boot App with AWS Java agent
+        run: |
+          agent_jar_name=`ls aws-opentelemetry-agent-*`
+          java -javaagent:$agent_jar_name -jar springboot-1.0.jar &
```

Comment:
I'm a bit surprised we're running directly without docker. But I guess it's ok if it works

Reply:
I am planning to add docker. This is just and initial draft where I wanted to see sample apps are up and running as expected.